orbit/pkg/useraction/mdm_migration_darwin.go

```diff
@@ -299,6 +299,7 @@ func (m *swiftDialogMDMMigrator) Show() error {
 		if err := m.renderMigration(); err != nil {
 			return fmt.Errorf("show: %w", err)
 		}
+		m.lastShown = time.Now()
```

Comment:
It's hard to tell without the larger call-flow context, but it looks like this could be a race with the read/write of `lastShown` in `ShowInterval` below? That is, it's not the same mutex that protects both accesses, so both could allow access at the same time?

Looking at the struct definition for `showMu`, it says:
```
// ensures only one dialog is open at a time, protects access to
// lastShown
```

but `showMu` is not locked when `lastShown` is read and (possibly) written in `ShowInterval`.

Reply:
you're right! my intent was to remove line `316` below since `Show` is being called by `ShowInterval`. Sanity checking, what do you think?

Reply:
There's still the read on line 311 below, though. If that becomes too complex to adjust, an option would be to store the Unix time in an atomic Int64 via [Store](https://pkg.go.dev/sync/atomic#Int64.Store) to set and [Load](https://pkg.go.dev/sync/atomic#Int64.Load) to read (and then re-create the `time.Time` from the unix timestamp to make the `time.Since` comparison).

Reply:
Oh, looking at it a bit closer, looks like `ShowInterval` as a whole could be integrated in `Show` and `intervalMu` being removed? Unless it's used elsewhere?

Reply:
thanks so much, indeed there's something fishy/redundant going on here. We do need both `Show` (always shows the dialog, no matter wat) and `ShowInterval` (shows the dialog only if the dialog wasn't open during the interval)

but I'm wondering if we need both locks, definitely room to make this cleaner. Will see if I can come up with something.

Reply:
Let me know if I can be of any help!

Reply:
@mna thank you! I would appreciate feedback on the changes I just pushed 🙏 

thanks as always for your help with the always tricky concurrency stuff.